pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -519,7 +522,7 @@ func (r *RepoSyncReconciler) watchConfigMaps(rs *v1beta1.RepoSync) error {
 		klog.Infoln("Adding watch for ConfigMaps in namespace ", rs.Namespace)
 		ctrlr := *r.controller
 
-		if err := ctrlr.Watch(&source.Kind{Type: withNamespace(&corev1.ConfigMap{}, rs.Namespace)},
+		if err := ctrlr.Watch(source.Kind(r.cache, withNamespace(&corev1.ConfigMap{}, rs.Namespace)),
```

Comment:
We didn't use cache for watches before. I think this also helps make watches more efficient.